### PR TITLE
LPS-131281 Add actions to Object APIs

### PR DESCRIPTION
**Pull request by javierdearcos**
Added actions to the Object Admin API endpoints, ObjectDefinition and ObjectField entities, and to the object entries. 

To have a cleaner actions implementation and ensure a consistent usage of APIs and Services, I've added permission logic to ObjectFields and also added some new permissions to the object entry resource permission template. This way, this logic is called by the actions infrastructure created in Vulcan and we don't have inconsistencies between the APIs and the services.

A sample of an ObjectDefinitionsPage get request:
```
{
  "actions": {
    "get": {
      "method": "GET",
      "href": "http://localhost:8080/o/object-admin/v1.0/object-definitions"
    },
    "create": {
      "method": "POST",
      "href": "http://localhost:8080/o/object-admin/v1.0/object-definitions"
    }
  },
  "facets": [],
  "items": [
    {
      "actions": {
        "get": {
          "method": "GET",
          "href": "http://localhost:8080/o/object-admin/v1.0/object-definitions/38404"
        }
      },...
]}
```
A sample of an ObjectDefinition ObjectFields get request:
```
{
  "actions": {
    "get": {
      "method": "GET",
      "href": "http://localhost:8080/o/object-admin/v1.0/object-definitions/38404/object-fields"
    },
    "create": {
      "method": "POST",
      "href": "http://localhost:8080/o/object-admin/v1.0/object-definitions/38404/object-fields"
    }
  },
  "facets": [],
  "items": [
    {
      "actions": {
        "get": {
          "method": "GET",
          "href": "http://localhost:8080/o/object-admin/v1.0/object-fields/38405"
        },
        "update": {
          "method": "PUT",
          "href": "http://localhost:8080/o/object-admin/v1.0/object-fields/38405"
        }
      },
      "id": 38405,...
]}
```

A sample for an ObjectEntry page get request:
```
{
  "actions": {
    "get": {
      "method": "GET",
      "href": "http://localhost:8080/o/c_sampleobjectdefinitions/"
    },
    "create": {
      "method": "POST",
      "href": "http://localhost:8080/o/c_sampleobjectdefinitions/"
    }
  },
  "facets": [],
  "items": [
    {
      "actions": {
        "get": {
          "method": "GET",
          "href": "http://localhost:8080/o/c_sampleobjectdefinitions/42197"
        },
        "update": {
          "method": "PUT",
          "href": "http://localhost:8080/o/c_sampleobjectdefinitions/42197"
        },
        "delete": {
          "method": "DELETE",
          "href": "http://localhost:8080/o/c_sampleobjectdefinitions/42197"
        }
      },
      "creator": {
        "additionalName": "",
        "contentType": "UserAccount",
        "familyName": "Test",
        "givenName": "Test",
        "id": 20127,
        "name": "Test Test"
      },
      "dateCreated": "2021-09-10T08:40:33Z",
      "dateModified": "2021-09-10T08:40:33Z",
      "externalReferenceCode": "",
      "id": 42197,
      "c_sampleObjectDefinitionId": 42197,
      "how": 180.5,
      "item": 5,
      "able": 10,
      "george": "The unsearchable brown fox jumps over the lazy dog. 0",
      "jig": 45,
      "baker": true,
      "dog": "The quick brown fox jumps over the lazy dog. 0!",
      "easy": "test0",
      "fox": "The english brown fox trusted the lazy dog. 0!",
      "charlie": "2021-09-10T08:40:33Z"
    },...
]}
```

I've needed to use an ActionUtil.addAction deprecated method. Maybe we can remove the deprecation, as this method it is also used in all the generated Base...ResourceImpl generation classes.

### CODE REVIEW

File: modules/apps/object/object-service/src/main/resources/resource-actions/resource-actions.xml.tpl [hunk 1]
```diff
@@ -38,6 +38,7 @@
 			<supports>
 				<action-key>ADD_OBJECT_ENTRY</action-key>
 				<action-key>PERMISSIONS</action-key>
+				<action-key>VIEW</action-key>
```
Comment (nhpatt) [inline modules/apps/object/object-service/src/main/resources/resource-actions/resource-actions.xml.tpl:41]: https://github.com/javierdearcos/liferay-portal/commit/c346c27a41d8cf0872ebb44a72be357b621a1d7d#diff-bd58fc68dd9af9e4f882dd3016fff2041dd1d4ddcaff06c92c2ccaa404fddae9 > I don't know anything about this.

**Review by nhpatt — CHANGES_REQUESTED** [review]
Can you rebase? there are some merge issues in the PR (including << in the code).

I don't know anything about permissions, I just try things until they work.... can you check Brian's commit?

File: modules/apps/object/object-service/src/main/resources/resource-actions/resource-actions.xml.tpl (hunk 1 above)
Comment (javierdearcos) [reply modules/apps/object/object-service/src/main/resources/resource-actions/resource-actions.xml.tpl:41]: Oh... Good to know. Thank you!

I supposed that the VIEW should be a default permission. Changing it and rebasing

**Review by nhpatt — APPROVED** [review]

**Comment by nhpatt** [comment]
LGTM, rebased and retrying ci:forward

**Comment by nhpatt** [comment]
ci:forward

**Comment by nhpatt** [comment]
ci:test:sf

**Comment by javierdearcos** [comment]
ci:forward

**Comment by nhpatt** [comment]
ci:forward

**Comment by nhpatt** [comment]
ci:forward

**Comment by javierdearcos** [comment]
I'll rebase and check the tests locally

**Comment by javierdearcos** [comment]
ci:forward

**Comment by javierdearcos** [comment]
ci:test:relevant

**Comment by nhpatt** [comment]
Sent directly here: https://github.com/brianchandotcom/liferay-portal/pull/107322

**Comment by nhpatt** [comment]
ci:close

**Pull request closed without merging**